### PR TITLE
Fixed user enumeration problem in SSPR by removing wording that contains masked email

### PR DESCRIPTION
**Pull request by hanyuanzhang-okta**
## Description:
- Fixed user enumeration problem in SSPR by removing the if condition that displays the wording with masked email if emails exists.
- The problem happens when going to /signin/forgot-password directly with a valid user identifier, or when adding/removing a phone authenticator in settings

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [x] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-525791](https://oktainc.atlassian.net/browse/OKTA-525791)

### Reviewers:
@okta/ciamx @jaredperreault-okta 
### Screenshot/Video:
Please note that there is a known SIW rendering issue that causes the extra spaces that is not related to this PR
NoMaskedEmailInAuthenticatorEnrollment https://okta.box.com/s/nyp77d2puq6sdy0mz3yhj82eiudis7e0
NoMaskedEmailInSSPR https://okta.box.com/s/oyme05nu5r18354j2xf7t48v4kyr7n0a

### Downstream Monolith Build:
https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-442c6d7-6375782a&page=1&pageSize=6&sha=d98454d0d6f180a484e64d525b7bc6edb79fcdd3&tab=main

### CODE REVIEW

**Review by justinabrokwah-okta — COMMENTED** [review]
LGTM! If possible can you add some screenshots or a video to verify the change please?

**Comment by hanyuanzhang-okta** [comment]
> LGTM! If possible can you add some screenshots or a video to verify the change please?
Thanks for the quick review (still a draft PR :D). I was gonna ask SIW folks to review it and add video after my local is up (I wanted to wait until the known SIW rendering issue is merged).

**Review by lokeshhemrajani-okta — COMMENTED** [review]
1. If `oie.email.verify.alternate.magicLinkToEmailAddress` and `oie.email.verify.subtitle.text.with.email` is not being used anywhere, then lets clean up from properties as well
2. I see `test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js` fixes the tests but I there are already other test case which are covering without email in the message. So, I would suggest cleanup tests rather than fixing it to avoid duplication.

**Comment by hanyuanzhang-okta** [comment]
> 1. If `oie.email.verify.alternate.magicLinkToEmailAddress` and `oie.email.verify.subtitle.text.with.email` is not being used anywhere, then lets clean up from properties as well
> 2. I see `test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js` fixes the tests but there are already other test case which are covering without email in the message. So, I would suggest cleanup tests rather than fixing it to avoid duplication.

I will remove them from properties file, but I do not think the tests needs clean up.
It is true that multiple ones are asserting on the property, but they are testing different scenarios. For example, 3 tests in `ChallengeAuthenticatorEmail_spec` are asserting on `Send a verification email by clicking on "Send me an email"`: one is the normal path (has valid profile), the other two covers scenarios of empty profile and no profile.

**Review by justinabrokwah-okta — APPROVED** [review]
Nice work! 🚀

**Comment by lokeshhemrajani-okta** [comment]
> > 1. If `oie.email.verify.alternate.magicLinkToEmailAddress` and `oie.email.verify.subtitle.text.with.email` is not being used anywhere, then lets clean up from properties as well
> > 2. I see `test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js` fixes the tests but there are already other test case which are covering without email in the message. So, I would suggest cleanup tests rather than fixing it to avoid duplication.
> 
> @lokeshhemrajani-okta I will remove them from properties file, but I do not think the tests needs clean up. It is true that multiple ones are asserting on the property, but they are testing different scenarios. For example, 3 tests in `ChallengeAuthenticatorEmail_spec` are asserting on `Send a verification email by clicking on "Send me an email"`: one is the normal path (has valid profile), the other two covers scenarios of empty profile and no profile.

Post your changes. `test.requestHooks(validOTPmock)('challenge email authenticator screen has right labels',` is same as `test.requestHooks(validOTPmockNoProfile)('challenge email authenticator screen has right labels when profile is null'`. So, you kind of dont need both of them because irrespective of profile present or not as we are not using anything from it now, you're displaying 1 message only. This applies to others as well.

**Comment by hanyuanzhang-okta** [comment]
> 

It is true that the value in profile is no longer used, but they are still different scenarios (user identified vs user not identified), I think we should keep them.

**Comment by hanyuanzhang-okta** [comment]
> > > 1. If `oie.email.verify.alternate.magicLinkToEmailAddress` and `oie.email.verify.subtitle.text.with.email` is not being used anywhere, then lets clean up from properties as well
> > > 2. I see `test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js` fixes the tests but there are already other test case which are covering without email in the message. So, I would suggest cleanup tests rather than fixing it to avoid duplication.
> > 
> > 
> > @lokeshhemrajani-okta I will remove them from properties file, but I do not think the tests needs clean up. It is true that multiple ones are asserting on the property, but they are testing different scenarios. For example, 3 tests in `ChallengeAuthenticatorEmail_spec` are asserting on `Send a verification email by clicking on "Send me an email"`: one is the normal path (has valid profile), the other two covers scenarios of empty profile and no profile.
> 
> Post your changes. `test.requestHooks(validOTPmock)('challenge email authenticator screen has right labels',` is same as `test.requestHooks(validOTPmockNoProfile)('challenge email authenticator screen has right labels when profile is null'`. So, you kind of dont need both of them because irrespective of profile present or not as we are not using anything from it now, you're displaying 1 message only. This applies to others as well.

It is true that the value in profile is no longer used, but they are still different scenarios (user identified vs user not identified), I think we should keep them.

**Review by jaredperreault-okta — APPROVED** [review]

**Comment by hanyuanzhang-okta** [comment]
Expected behavior updated, should fix this in backend. Declining this PR.

**Pull request closed without merging**
